src/hooks/data/vaults/useVault.ts

```diff
@@ -0,0 +1,41 @@
+import { Address, useContractReads } from "wagmi"
+
+import { VaultProps } from "@/lib/types"
+import useActiveChainId from "@/hooks/useActiveChainId"
+
+import { vaultCompounderAbi } from "@/constant/abi"
+
+// TODO: Use API first
+
+export function useVault({ asset, type, vaultAddress }: VaultProps) {
```

Comment:
Pass this the normalized data from `useConcentrator` or `useCompounder` and retrieve data from the ERC4626 vault contract.